### PR TITLE
Add support for authorized-networks

### PR DESCRIPTION
**Pull request by tun0**

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -1,3 +1,8 @@
+variable "authorized_networks" {
+  type = map(string)
```
Comment (EvyBongers) [inline variables.tf]: Maybe use `list(object{})` instead to support optional `expiration_time`?
```suggestion
  type = list(object({
    name = optional(string) # Not sure about optional. It's optional for the API, we might want to enforce it
    value = string
    expiration_time = optional(string)
  })
```
Alternatively:
```suggestion
  type = map(object({ # map key is CIDR
    name = optional(string) # Not sure about optional. It's optional for the API, we might want to enforce it
    expiration_time = optional(string)
  })
```

File: instance.tf [hunk 2]
```diff
@@ -27,6 +27,16 @@ resource "google_sql_database_instance" "instance" {
...
       ipv4_enabled = true
       require_ssl  = true
+
+      dynamic "authorized_networks" {
+        for_each = var.authorized_networks
+        iterator = network
+
+        content {
+          name  = network.key
+          value = network.value
```
Comment (EvyBongers) [inline instance.tf]: CIDR makes for a good unique key, while names do not
```suggestion
          name  = network.value
          value = network.key
```

**Review by EvyBongers — CHANGES_REQUESTED** [review]

File: variables.tf (hunk 1 above)
Comment (EvyBongers) [reply variables.tf]: To be clear, I'm okay with leaving it as it is. Just considering flexibility (or over engineering)

Comment (tun0) [reply variables.tf]: I actually had it using an `object` initially but changed into into something "simpler". Didn't think about the possibility of (future) additional parameters. Going back to `object` it is.

Comment (EvyBongers) [reply variables.tf]: Benefit is the first approach is user readability, benefit of the second approach is stripping the need to create a custom for loop (though that's exactly the kind of thing we want to stow away in a module).

Comment (tun0) [reply variables.tf]: Will postpone supporting `expiration_time` until we actually need it.

Comment (EvyBongers) [reply variables.tf]: Regarding the first proposal, maybe use `cidr` instead of `value` for a key, as it's more descriptive.

**Review by EvyBongers — APPROVED** [review]

**Pull request merged**